ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java

```diff
@@ -178,12 +198,21 @@ long getTotalFileSize() {
       return sizeInBytes;
     }
 
-    long getTotalCacheSize() {
-      return segments.stream().mapToLong(LogSegment::getTotalCacheSize).sum();
+    LogSegmentListCacheInfo getCacheInfo() {
+      long totalCacheSize = 0L;
+      long cachedCount = 0L;
+      for (LogSegment segment: segments) {
+        if (segment.hasCache()) {
+          cachedCount++;
+        }
+        // TODO: Should we only increment when hasCache = true?
+        totalCacheSize += segment.getTotalCacheSize();
+      }
+      return new LogSegmentListCacheInfo(totalCacheSize, cachedCount);
     }
 
     long countCached() {
-      return segments.stream().filter(LogSegment::hasCache).count();
+      return getCacheInfo().getCachedCount();
```

Comment:
Let's keep the original code for `raftLogMetrics`.  It only needs the count.

Reply:
Sure, reverted to the original code.